app/models/host/base.rb

```diff
@@ -349,11 +349,13 @@ def get_interface_scope(name, attributes, base = self.interfaces)
           rescue Net::Validations::Error
             logger.debug "invalid mac during parsing: #{attributes[:macaddress]}"
           end
-          base = base.where(:mac => macaddress)
+
+          nics = base.where(:mac => macaddress)
```

Comment:
I find this a bit more readable (not combining if in else branch)

```ruby
mac_based = base.where(:mac => macaddress)
if attributes[:virtual]
  mac_based.virtual.where(:identifier => name)
elsif !self.managed?
  base.where(:identifier => name)
else
  mac_based.physical
end
```

one difference is that I don't even try `nics` and fallback to base which I think is also good, feel free to add it to elsif if you disagree

also please add a comment on line 343 why we handle unmanaged hosts differently